src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -430,8 +430,10 @@ public final MemorySessionImpl sessionImpl() {
     }
 
     private IndexOutOfBoundsException outOfBoundException(long offset, long length) {
-        return new IndexOutOfBoundsException(String.format("Out of bound access on segment %s; new offset = %d; new length = %d",
-                        this, offset, length));
+        return new IndexOutOfBoundsException(String.format("Out of bound access on segment %s; " +
```

Comment:
The new message looks ok, but note that this message is also reused by the `asSlice` methods. If we want to clarify the messages, perhaps we should think about rearranging the code a bit?

Related note: this is also why the routine that check OOB access needs to worry about `length` being zero -- because that's a valid value for `asSlice` (but not for access).

Reply:
> length being zero -- because that's a valid value for asSlice (but not for access).

does it mean `checkBounds(...)` version in `master` works incorrectly for access now? Since length == 0 is not treated as an error.